main.js

```diff
@@ -14,77 +19,88 @@ document.addEventListener("DOMContentLoaded", function () {
     if (newItemInput.value.length > 0) {
-      createNewTodoItem(newItemInput.value);
+      addTask(newItemInput.value, false);
       newItemInput.value = "";
     }
   });
 
-  newItemInput.addEventListener("keypress", (e) => {
-    if (e.charCode === 13 && newItemInput.value.length > 0) {
-      createNewTodoItem(newItemInput.value);
+  // Add new task on Enter key press
+  newItemInput.addEventListener("keypress", (event) => {
+    if (event.charCode === 13 && newItemInput.value.length > 0) {
+      addTask(newItemInput.value, false);
       newItemInput.value = "";
     }
   });
 
-  function createNewTodoItem(text) {
-    const elem = document.createElement("li");
-    elem.classList.add("flex-row");
-
-    elem.innerHTML = `
-        <label class="list-item">
-          <input type="checkbox" name="todoItem">
-          <span class="checkmark"></span>
-          <span class="text">${text}</span>
-        </label>
-        <span class="remove"></span>
-      `;
-
-    if (
-      document.querySelector('.filter input[type="radio"]:checked').id ===
-      "completed"
-    ) {
-      elem.classList.add("hidden");
-    }
-
-    todoList.append(elem);
-    updateItemsCount(1);
+  function addTask(text, isChecked) {
+    const task = { text, isChecked };
+    tasks.push(task);
+    createNewTodoItem();
+    saveTasks();
   }
 
-  function updateItemsCount(number) {
-    itemsLeft.innerText = +itemsLeft.innerText + number;
-  }
+  function createNewTodoItem() {
+    todoList.innerHTML = "";
+    tasks.forEach((task, index) => {
+      const elem = document.createElement("li");
+      elem.classList.add("flex-row");
+
+      elem.innerHTML = `
+          <label class="list-item">
+            <input type="checkbox" name="todoItem" ${
+              task.isChecked ? "checked" : ""
+            } data-index="${index}">
+            <span class="checkmark"></span>
+            <span class="text">${task.text}</span>
+            </label>
+            <button class="edit" data-index="${index}">Edit</button>
+          <span class="remove" data-index="${index}"></span>
+        `;
+
+      if (
+        document.querySelector('.filter input[type="radio"]:checked').id ===
+          "completed" &&
+        !task.isChecked
+      ) {
+        elem.classList.add("hidden");
+      } else if (
+        document.querySelector('.filter input[type="radio"]:checked').id ===
+          "active" &&
+        task.isChecked
+      ) {
+        elem.classList.add("hidden");
+      }
 
-  function removeTodoItem(elem) {
-    elem.remove();
-    updateItemsCount(-1);
+      todoList.append(elem);
+    });
+    itemsLeft.innerText = tasks.filter((task) => !task.isChecked).length;
   }
 
-  todoList.addEventListener("click", (event) => {
-    if (event.target.classList.contains("remove")) {
-      removeTodoItem(event.target.parentElement);
-    }
-  });
+  // Remove a to-do item
+  function removeTask(index) {
+    tasks.splice(index, 1);
+    createNewTodoItem();
```

Comment:
instead of calling this function, you should add Observer to `tasks`, see again https://css-tricks.com/build-a-state-management-system-with-vanilla-javascript/

Reply:
Try to implement exactly the same architecture that was mentioned in article